compiler/optimizer/Inliner.cpp

```diff
@@ -1790,7 +1790,11 @@ TR_InlinerBase::addGuardForVirtual(
       {
       createdHCRAndVirtualGuard = true;
 
+      // we merge virtual guards and OSR guards for simplicity in most modes
+      // when using OSR to implement HCR we keep the HCR guards distinct since they
+      // will undergo special processing later in the compilation
       if (virtualGuard &&
+          comp()->getHCRMode() != TR::osr &&
```

Comment:
I don't think I agree with that - what we want the inliner to generate and what is supported are two different things. Once the special processing of HCR guards is done they could be remerged to reduce the number of nodes so saying the platform doesn't support merging is different than saying we don't want to generate merged guards immediately in my opinion.